Sources/PointFree/Account/Change.swift

```diff
@@ -108,8 +98,24 @@ func requireActiveSubscription<A>(
       <| middleware
 }
 
+func subscriptionModificationErrorMiddleware<A>(
+  _ conn: Conn<StatusLineOpen, A>
+  ) -> IO<Conn<ResponseEnded, Data>> {
+
+  return conn |> redirect(
+    to: .account(.index),
+    headersMiddleware: flash(
+      .error,
+      """
+      We couldn’t modify your subscription at this time. Please try again or contact
+      <support@pointfree.co>.
+      """
+    )
+  )
+}
+
 private func requireValidSeating(
-  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, Tuple4<Stripe.Subscription, User, Int, Pricing>, Data>
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, (Stripe.Subscription, Pricing), Data>
```

Comment:
this middleware was dealing with more data than necessary so I focused it a bit more. this turned out to not really be necessary (I thought I was gonna need to share this at one point), but thought I might as well keep it in

Reply:
interesting change away from Tuple. in general do you think we should move away from the structural types?